test/Generators/Microsoft.Gen.Logging/Unit/TypeSymbolExtensionsTests.cs

```diff
@@ -0,0 +1,320 @@
+                        static partial void M1(ILogger logger, {typeReference} property);
+                    }}
+                }}";
+
+        // Create compilation and extract symbols
+        Compilation compilation = CompilationHelper.CreateCompilation(source);
+        SymbolHolder? symbolHolder = SymbolLoader.LoadSymbols(compilation, _diagCallback);
+
+        IEnumerable<ISymbol> methodSymbols = compilation.GetSymbolsWithName("M1", SymbolFilter.Member);
+
+        // Assert
+        Assert.NotNull(symbolHolder);
+        ISymbol symbol = Assert.Single(methodSymbols);
+        var methodSymbol = Assert.IsAssignableFrom<IMethodSymbol>(symbol);
+        var parameterSymbol = Assert.Single(methodSymbol.Parameters, p => p.Name == "property");
+
+        Assert.Equal(expectedResult, parameterSymbol.Type.IsSpecialType(symbolHolder));
+    }
+
+    [Theory]
+    [InlineData("ToString", "Test", true)]
+    [InlineData("RandomMethod", "Test", false)]
+    [InlineData("ToooooString", "Test", false)]
+    public void ValidateHasCustomToString(string methodName, string typeReference, bool expectedResult)
+    {
+        // Generate the code
+        string source = $@"
+                namespace Test
+                {{
+                    using System;
+                    using Microsoft.Extensions.Logging;
+
+                    class {typeReference} 
+                    {{
+                        public override string {methodName}()
+                        {{
+                            throw new NotImplementedException();
+                        }}
+                    }}
+
+                    class NonConvertible {{ }}
+
+                    partial class C
+                    {{
+                        [LoggerMessage(EventId = 1, Level = LogLevel.Debug, Message = ""M1"")]
+                        static partial void M1(ILogger logger, {typeReference} property);
+
+                        public override string {methodName}()
+                        {{
+                            throw new NotImplementedException();
+                        }}
+                    }}
+                }}";
+
+        // Create compilation and extract symbols
+        Compilation compilation = CompilationHelper.CreateCompilation(source);
+        SymbolHolder? symbolHolder = SymbolLoader.LoadSymbols(compilation, _diagCallback);
+        IEnumerable<ISymbol> methodSymbols = compilation.GetSymbolsWithName("M1", SymbolFilter.Member);
+
+        // Assert
+        Assert.NotNull(symbolHolder);
+        ISymbol symbol = Assert.Single(methodSymbols);
+        var methodSymbol = Assert.IsAssignableFrom<IMethodSymbol>(symbol);
+        var parameterSymbol = Assert.Single(methodSymbol.Parameters, p => p.Name == "property");
+
+        Assert.Equal(expectedResult, parameterSymbol.Type.HasCustomToString());
+    }
+
+    [Fact]
+    public void GetPossiblyNullWrappedType_NonNullableType_ReturnsSameType()
```

Comment:
I'm a little confused about the assertions for the following tests. It doesn't seem to be validating the `GetPossiblyNullWrappedType` method unless I'm missing something obvious.

Reply:
`GetPossiblyNullWrappedType` returns either original ITypeSymbol param or the TypeArguments if that param is nullable and falls under multiple conditions. This specific test validates that the parameter is being returned if it doesn't meet those conditions

Reply:
Let's boil it down to what it means from the perspective of the type. A null wrapped type would look like: `Nullable<int>` or `int?`. Underneath it's a generic type, yes. But I think it would be more valid to check these scenarios:

- `Nullable<T>` returns `T`
- `T?` returns `T`
- `List<T>` returns `List<T>`
- `T` returns `T`

Reply:
Updated test cases for these options